### DISCUSSION TITLE
How to write SMILES for square-planar organometallic complex with cis trans stereochemistry?

### DISCUSSION THREAD

**Discussion by manajit-das** [rdkit/rdkit#5865] [Q&A]
I have the following SMILES string: 'CC(C)[N+](C(C)C)(C(C)C)[Pd](Cl)(c1ccccc1)[N+](C)(C)C' with no cis/trans information. 
This is obtained using AllChem.ReactionFromSmarts(), [I think currently it doesn't include stereo information, I tried to add!!]

However I want to get the SMILES string with the two N atoms trans to each other. One solution  available is that use of '@SP1/2/3' addition, like this : 
![demo](https://user-images.githubusercontent.com/59758360/207525148-9d6e5e70-805c-4853-abdd-a20135131187.PNG)

My problem is how to get the SMILES string with the two N atoms trans for multiple such cases without visually checking whether @SP1/2/3 is required?
Thank you so much.

**Comment by greglandrum** [comment]
@manajit-das : I don't understand the question.
The way that you express the orientation of ligands around a square planar center is using the `@SP1`, `@SP2`, `@SP3` markers in SMILES. If you don't provide that information in the SMILES then the RDKit has no way of knowing if you mean tetrahedral, square planar, etc.

If you have questions about what those non-tetrahedral stereo labels mean, you can check the RDKit documentation:
https://www.rdkit.org/docs/RDKit_Book.html#smiles-notation

**Reply by manajit-das** [reply]
Hi, Sorry 
Actually I want in the reaction template (given below here), so that the two 'N' atoms are trans to each other. Can I pass such information into the reaction template, so that all the product has trans relationship between the 'N' atoms.
![reaction2](https://user-images.githubusercontent.com/59758360/208238890-bc0cb06e-ad26-4d21-946e-c2ab3eec4ed3.PNG)

**Reply by HelloJocelynLu** [reply]
Hi Greg,

Apologies for revisiting the past conversation. I encountered a similar issue: Can a chiral tag like `@SP1`, `@SP2` be programmatically constructed in rdkit? I can generate a generic "SP" chiral tag. For instance:
```python
center = Chem.MolFromSmiles("[Pd+2]")
center.GetAtomWithIdx(0).SetChiralTag(Chem.ChiralType.CHI_SQUAREPLANAR)
print(Chem.MolToSmiles(center)) # [Pd@SP+2]
```
However, I aim to achieve [Pd@SP1+2]. Is it feasible to accomplish this? Currently, I am using `str.replace` on SMILES to achieve this...
